torch/fx/graph.py

```diff
@@ -131,12 +135,29 @@ def _name(self, target: Target) -> str:
     def python_code(self, root_module: str) -> Tuple[str, str, List[str]]:
         free_vars: List[str] = []
         body: List[str] = []
+
```

Comment:
yeah, I think it makes sense to tag the nodes, but then we should either (1, preferred) delete the information when writing the code, or (2) reconstruct the forward methods on the submodules as they were originally. The context manager stuff seems like a poor compromise between them. Its nearly 1/3 of the entire code formatting code and then it makes the resulting code depend on a context manager that is going to be hard to script. (2) seems like it is nearly already implemented since there is a utility function to split up the graph module into multiple modules already.

Reply:
I don't like option (1) since it means that the module structure information is lost after the first source-to-source transformation. One example use case we're trying to solve here is e.g. when someone applies quantization to the whole model, then wants to lower only a subset to an accelerator. This would require preserving the information even through the quantization transforms so that it can be used later to outline the proper subgraph for lowering

Not sure if I particularly like option (2). My concern is that there's too much implicit magic going on in the Graph to code conversion process, and potentially we're putting more constraints on the structure of the `module_qualname` field that people generating Graphs need to deal with. I can write out a prototype for this, though, and we can see